crates/iceberg/src/catalog.rs

```diff
@@ -0,0 +1,163 @@
+use crate::Result;
+use async_trait::async_trait;
+use std::collections::HashMap;
+
+/// The catalog API for Iceberg Rust.
+#[async_trait]
+pub trait Catalog {
+    /// Get the catalog configuration for specified warehouse.
+    async fn get_config(&self, warehouse: Option<&str>) -> Result<CatalogConfig>;
+
+    /// List namespaces from table.
+    async fn list_namespaces(&self, parent: Option<NamespaceIdent>) -> Result<Vec<NamespaceIdent>>;
+
+    /// Create a new namespace inside the catalog.
+    async fn create_namespace(
+        &self,
+        namespace: NamespaceIdent,
+        properties: HashMap<String, String>,
+    ) -> Result<Namespace>;
+
+    /// Get a namespace information from the catalog.
+    async fn get_namespace(&self, namespace: NamespaceIdent) -> Result<Namespace>;
+
+    /// Update a namespace inside the catalog.
+    ///
+    /// # Behavior
+    ///
+    /// The properties must be the full set of namespace.
+    async fn update_namespace(
+        &self,
+        namespace: NamespaceIdent,
+        properties: HashMap<String, String>,
+    ) -> Result<()>;
+
+    /// Drop a namespace from the catalog.
+    async fn drop_namespace(&self, namespace: NamespaceIdent) -> Result<()>;
+
+    /// List tables from namespace.
+    async fn list_tables(&self, namespace: NamespaceIdent) -> Result<Vec<TableIdent>>;
+
+    /// Create a new table inside the namespace.
+    async fn create_table(
+        &self,
+        namespace: NamespaceIdent,
+        creation: TableCreation,
+    ) -> Result<Table>;
+
+    /// Load table from the catalog.
+    async fn load_table(&self, table: TableIdent) -> Result<Table>;
+
+    /// Drop a table from the catalog.
+    async fn drop_table(&self, table: TableIdent) -> Result<()>;
+
+    /// Check if a table exists in the catalog.
+    async fn stat_table(&self, table: TableIdent) -> Result<bool>;
+
+    /// Rename a table in the catalog.
+    async fn rename_table(&self, src: TableIdent, dest: TableIdent) -> Result<()>;
+
+    /// Commit a table to the catalog.
+    async fn commit_table(&self, table: TableIdent, commit: TableCommit) -> Result<Table>;
+
+    /// Commit multiple tables to the catalog as an atomic operation.
+    async fn commit_tables(&self, tables: Vec<(TableIdent, TableCommit)>) -> Result<()>;
+}
+
+/// The config the catalog.
+pub struct CatalogConfig {
+    defaults: HashMap<String, String>,
+    overrides: HashMap<String, String>,
+}
+
+/// NamespaceIdent represents the identifier of a namespace in the catalog.
+pub struct NamespaceIdent(Vec<String>);
+
+/// Namespace represents a namespace in the catalog.
+pub struct Namespace {
+    name: NamespaceIdent,
+    properties: HashMap<String, String>,
+}
+
+/// TableIdent represents the identifier of a table in the catalog.
+pub struct TableIdent {
+    namespace: NamespaceIdent,
+    name: String,
+}
+
+/// Table represents a table in the catalog.
+pub struct Table {
```

Comment:
Is this Table concept is same as Table in icelake (https://github.com/icelake-io/icelake/blob/848e8b77aa2f903b265e459a9fcf0777da022fa6/icelake/src/table.rs#L25)?🤔 I mean whether it will provide method like getting a task writer or it just represent the metadata.

Reply:
> I mean whether it will provide method like getting a task writer or it just represent the metadata.

Personally, I believe that they should be two separate structs.